### PR TITLE
[SYCL][ESIMD] Fix logic and arithmetic operators for simd_view with scalar

### PR DESCRIPTION
**Pull request by sndmitriev**
These operators should call corresponding operator for simd_obj_imp with
scalar instead of constructing simd object from input scalar themselves.
simd_obj_impl operators will do the necessary type promotion for operands
if needed which is not done in the current implementation.

### CODE REVIEW

File: sycl/test/esimd/simd_view_bin_op.cpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+#include <sycl/ext/intel/experimental/esimd.hpp>
+
+using namespace sycl::ext::intel::experimental::esimd;
+
+simd<short, 4> test1(int X, simd<short, 16> Y) __attribute__((sycl_device)) {
+  // CHECK-LABEL: test1
+  // CHECK-SAME: i32 [[X:%.+]],
+  // CHECK-NOT: trunc i32 [[X]] to i16
+  return Y.select<4, 1>(0) * X;
```
Comment (kychendev) [inline sycl/test/esimd/simd_view_bin_op.cpp:14]: Did you mean to test X * Y.select<4, 1>(0) ?

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/operators.hpp [hunk 2]
```diff
@@ -301,8 +300,7 @@ namespace __SEIEE {
             class = std::enable_if_t<                                          \
                 __SEIEED::is_any_simd_view_type_v<SimdViewT1> && COND>>        \
   inline auto operator BINOP(const SimdViewT1 &LHS, T2 RHS) {                  \
-    using SimdT = typename SimdViewT1::value_type;                             \
-    return LHS.read() BINOP SimdT(RHS);                                        \
+    return LHS.read() BINOP RHS;                                               \
```
Comment (kychendev) [inline sycl/include/sycl/ext/intel/experimental/esimd/detail/operators.hpp:303]: Similar issues need to be fixed for CMPOP implementation below.

Comment (kbobrovs) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/operators.hpp:303]: > Similar issues need to be fixed for CMPOP implementation below.

I suggest to do this in another PR.

File: sycl/test/esimd/simd_view_bin_op.cpp (hunk 1 above)
Comment (kbobrovs) [reply sycl/test/esimd/simd_view_bin_op.cpp:14]: right, otherwise test1 and test2 are the same

Comment (sndmitriev) [reply sycl/test/esimd/simd_view_bin_op.cpp:14]: Right, that was supposed to be X * Y.select<4, 1>(0).

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/operators.hpp (hunk 2 above)
Comment (sndmitriev) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/operators.hpp:303]: Yes, CMPOP needs a similar fix. I will do this change in a separate PR.

**Review by kbobrovs — APPROVED** [review]

**Review by kychendev — APPROVED** [review]

**Pull request merged**